tests/integration.rs

```diff
@@ -32,6 +34,50 @@ pub const TEST_BINS_OUT: &str = "bin";
 pub const TIMEOUT_SECS: u64 = 60 * 60;
 pub const MAX_ASSERT_ELEMENTS: usize = 100;
 
+/// `ENARX_BACKEND` environment variable value
+pub static ENARX_BACKEND: Lazy<Option<String>> = Lazy::new(|| match var("ENARX_BACKEND") {
+    Ok(backend) => Some(backend),
+    Err(VarError::NotUnicode(..)) => panic!("`ENARX_BACKEND` value is not valid unicode"),
+    Err(VarError::NotPresent) => None,
+});
+
+/// Returns `true` if SGX backend should be used by Enarx given the environment
+pub fn is_sgx() -> bool {
+    match ENARX_BACKEND.as_deref() {
+        Some("sgx") => true,
+        Some(..) => false,
+        None => cfg!(host_can_test_sgx),
+    }
+}
+
+/// Returns `true` if SEV backend should be used by Enarx given the environment
+pub fn is_sev() -> bool {
+    match ENARX_BACKEND.as_deref() {
+        Some("sev") => true,
+        Some(..) => false,
+        // NOTE: SGX backend is prioritized
+        None => cfg!(host_can_test_sev) && !is_sgx(),
+    }
+}
+
+/// Returns `true` if KVM backend should be used by Enarx given the environment
+pub fn is_kvm() -> bool {
+    match ENARX_BACKEND.as_deref() {
+        Some("kvm") => true,
+        Some(..) => false,
+        None => cfg!(host_can_test_kvm) && !is_sgx() && !is_sev(),
```

Comment:
Since this match arm means that every other `is_foo` function will take the `None` branch, and since `is_sev`'s `None` branch calls `!is_sgx`, I believe the call to `!is_sgx()` is redundant work on this line.